### DISCUSSION TITLE
Apply any module to specific class

### DISCUSSION THREAD

**Discussion by ManosMachairas** [checkstyle/checkstyle#14331] [Q&A] [answered]
Hi,
I would like to ask if there is any way to apply an existing rule to a specific class. For example, let's say that I want to apply "RegexpSingleline" module to Enum classes only. I imagine it like this, but property "tokens" cannot be used in this module and it results in error:

![image](https://github.com/checkstyle/checkstyle/assets/46612530/225775c2-5328-4a21-877f-d7d50af56d13)

Is there any way to achieve it without creating a custom rule?
Thanks in advance!!!

**Comment by nrmancuso** [comment, marked as answer]
@ManosMachairas there is a few different ways this could be handled, but it might be better to provide a concrete example of what you want to achieve, such as “How can I enforce xyz naming convention on enum constants?”.

**Reply by ManosMachairas** [reply]
A rule I want to apply is exactly this one! Enum entries should follow the following format: "^[A-Z][A-Z0-9_]*$".

At first, I tried to create a "TreeWalker" and apply the "tokens" property in it, so that each module inherit that property and check only enum classes like the snippet below, but "TreeWalker" does not accept this property.

![image](https://github.com/checkstyle/checkstyle/assets/46612530/bc1efd25-d7fc-4440-ad18-f7964297ebdb)

**Reply by nrmancuso** [reply]
Please take a look through our naming conventions checks: https://checkstyle.sourceforge.io/checks/naming/index.html

These are checks that target specific language elements in Java code, whereas regexp checks  are more general.

I also recommend you to read our documentation about configuring checkstyle: https://checkstyle.sourceforge.io/config.html

Also, it is helpful to provide examples of code that you expect to be acceptable, and code that you expect violations on, to make sure we understand your needs.

**Reply by romani** [reply]
Use https://checkstyle.org/checks/naming/typename.html
And put enum token only.

You need to put config under proper parent, it should be Checker in your case
https://checkstyle.org/checks/regexp/regexpsingleline.html#Example1-config
